plutus-playground-client/src/Types.purs

```diff
@@ -168,6 +171,42 @@ instance showDragAndDropEventType :: Show DragAndDropEventType where
+  toEvent CompileProgram = Just $ defaultEvent "CompileProgram"
+  toEvent (HandleBalancesChartMessage _) = Nothing
+  toEvent CheckAuthStatus = Nothing
+  toEvent (GistAction PublishGist) = Just $ (defaultEvent "Publish") { category = Just "Gist" }
+  toEvent (GistAction (SetGistUrl _)) = Nothing
+  toEvent (GistAction LoadGist) = Just $ (defaultEvent "LoadGist") { category = Just "Gist" }
+  toEvent (GistAction (AjaxErrorPaneAction _)) = Nothing
+  toEvent (ChangeView view) = Just $ (defaultEvent "View") { label = Just $ show view }
+  toEvent (LoadScript script) = Just $ (defaultEvent "LoadScript") { label = Just script }
+  toEvent AddSimulationSlot = Just $ (defaultEvent "AddSimulationSlot") { category = Just "Simulation" }
+  toEvent (SetSimulationSlot _) = Just $ (defaultEvent "SetSimulationSlot") { category = Just "Simulation" }
+  toEvent (RemoveSimulationSlot _) = Just $ (defaultEvent "RemoveSimulationSlot") { category = Just "Simulation" }
+  toEvent (ModifyWallets AddWallet) = Just $ (defaultEvent "AddWallet") { category = Just "Wallet" }
+  toEvent (ModifyWallets (RemoveWallet _)) = Just $ (defaultEvent "RemoveWallet") { category = Just "Wallet" }
+  toEvent (ModifyWallets (ModifyBalance _ (SetBalance _ _ _))) = Just $ (defaultEvent "SetBalance") { category = Just "Wallet" }
+  toEvent (ActionDragAndDrop _ eventType _) = Just $ (defaultEvent (show eventType)) { category = Just "Action" }
+  toEvent EvaluateActions = Just $ (defaultEvent "EvaluateActions") { category = Just "Action" }
+  toEvent (ChangeSimulation (PopulateAction _ _)) = Just $ (defaultEvent "PopulateAction") { category = Just "Action" }
+  toEvent (ChangeSimulation (ModifyActions (AddAction _))) = Just $ (defaultEvent "AddAction") { category = Just "Action" }
+  toEvent (ChangeSimulation (ModifyActions (AddWaitAction _))) = Just $ (defaultEvent "AddWaitAction") { category = Just "Action" }
+  toEvent (ChangeSimulation (ModifyActions (RemoveAction _))) = Just $ (defaultEvent "RemoveAction") { category = Just "Action" }
+  toEvent (ChangeSimulation (ModifyActions (SetPayToWalletValue _ _))) = Just $ (defaultEvent "SetPayToWalletValue") { category = Just "Action" }
+  toEvent (ChangeSimulation (ModifyActions (SetPayToWalletRecipient _ _))) = Just $ (defaultEvent "SetPayToWalletRecipient") { category = Just "Action" }
+  toEvent (ChangeSimulation (ModifyActions (SetWaitTime _ _))) = Just $ (defaultEvent "SetWaitTime") { category = Just "Action" }
+  toEvent (ChangeSimulation (ModifyActions (SetWaitUntilTime _ _))) = Just $ (defaultEvent "SetWaitUntilTime") { category = Just "Action" }
+  toEvent (ChainAction (FocusTx (Just _))) = Just $ (defaultEvent "BlockchainFocus") { category = Just "Transaction" }
+  toEvent (ChainAction (FocusTx Nothing)) = Nothing
+  toEvent (ChainAction (ClipboardAction (Clipboard.CopyToClipboard _))) = Just $ (defaultEvent "ClipboardAction") { category = Just "CopyToClipboard" }
```

Comment:
It'd be really nice to break this down further now. So for example `toEvent (ChangeSimulation subaction) = toEvent subaction`.

But that's probably best in another PR.

Reply:
Ah, yes, good idea. And the IsEvent instances for the subactions should probably be in the other relevant type modules. In which case, best leave this until after we've got a separate Simulator.Types module - which is still the plan, right?

If we put the IsEvent GistAction instance in `web-common/src/Gists/Types.purs` (where arguably it would belong), then we'd be regimenting the same gist action tracking in both playgrounds. Would that be a good thing or a bad thing? We're not likely to want to track different things, are we?